crit/mempages.go

```diff
@@ -0,0 +1,145 @@
+package crit
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/checkpoint-restore/go-criu/v6/crit/images/pagemap"
+)
+
+var sysPageSize = uint64(syscall.Getpagesize())
+
+// MemoryReader is a struct used to retrieve
+// the content of memory associated with a specific process ID (pid).
+// New instances should be created with NewMemoryReader()
+type MemoryReader struct {
+	checkpointDir  string
+	pid            uint32
+	pagesID        uint32
+	pageSize       uint64
```

Comment:
I think we should use the type we get from `Getpagesize()`. Which is just `int`. Not sure it makes sense to cast it to another value.